lumigator/python/mzai/backend/backend/tests/api/routes/test_health.py

```diff
@@ -60,3 +83,45 @@ def test_get_job_metadata_ok(app_client: TestClient,
     actual = json.loads(response.text)
     expected = load_json(json_data_health_job_metadata_ok)
     assert expected == actual
+
+def test_job_logs(app_client: TestClient, valid_experiment_dataset: str):
```

Comment:
I think this is fine for an integration test, but perhaps here we'd better be focusing on the individual functionality without uploading datasets or running actual jobs. Two mock responses, one for a missing job and another one for an existing one, should probably be enough. You can validate both wrt `JobLogsResponse` and assert you are getting the proper status codes.

Reply:
This might be useful for mocking the return data from Ray to exercise the cases you want in the route code:

https://github.com/mozilla-ai/lumigator/pull/318/files#diff-199b80c426fab13e604600bdb49d196d54459bc0b34bf6d027ed49b80ee4081cR23-R26

Reply:
I think that integration testing is more appropriate here than a basic test, since we're dealing with a third party whose behaviour may change at some point. That said, I'll include also a basic test with mocked requests for completion.

_That said_, I think we should open an issue to separate basic tests from integration tests, and execute them in separate ways so that the external services need not be started always, and the basic tests are executed first for faster failures. Wdyt?

Reply:
Can't we have both (unit and integration tests)? 😄

Reply:
My bad. I didn't push my latest work :-/ Yes, the plan was to have both (and later on separate basic vs integration).